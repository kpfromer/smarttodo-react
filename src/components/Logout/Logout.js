import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { logout } from "../../actions/login";
import { Redirect } from "react-router-dom";

export class Logout extends PureComponent {

  static propTypes = {
    logout: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.logout();
  }

  render() {
    return <Redirect push to="/" />;
  }
}

export default connect(null, { logout })(Logout);