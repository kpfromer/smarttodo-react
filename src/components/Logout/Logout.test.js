import { SetupComponent } from "react-component-setup";
import { Logout } from "./Logout";

const { shallow: setup } = SetupComponent({
  Component: Logout
});

describe('Logout', () => {
  let wrapper, mockLogout;

  beforeEach(() => {
    mockLogout = jest.fn();
    ({ wrapper } = setup({
      logout: mockLogout
    }))
  });

  it('logouts user', () => {
    expect(mockLogout).toHaveBeenCalled();
  });
  it('redirects to /', () => {
    expect(wrapper).toMatchSnapshot();
  });
});