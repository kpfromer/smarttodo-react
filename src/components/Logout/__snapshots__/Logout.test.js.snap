// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout redirects to / 1`] = `
<Redirect
  push={true}
  to="/"
/>
`;
